addons/website_event_sale/controllers/main.py

```diff
@@ -70,3 +71,19 @@ def registration_confirm(self, event, **post):
                 request.website.sale_reset()
 
         return res
+
+
+class PaymentPortalOnsite(PaymentPortal):
+
+    def _validate_transaction_for_order(self, transaction, sale_order_id):
+        """
+        Throws a ValidationError if the user tries to pay for a ticket which isn't available
+        """
+        super()._validate_transaction_for_order(transaction, sale_order_id)
+        sale_order_sudo = request.env['sale.order'].browse(sale_order_id).exists().sudo()
+        count_per_ticket = request.env['event.registration']._read_group(
+            [('sale_order_id', 'in', sale_order_sudo.ids), ('state', '!=', 'cancel')],
+            ['event_ticket_id'], ['__count']
+        )
```

Comment:
Why not just use `ticket.seats_available`? Would be safer as for example I don't think we consider "draft" as a taken seat, ever

Reply:
Hello, here I'm counting the number of ticket per events that the user intends to purchase, then for each ticket I check that we have enough ticket to cover the purchase. I looked at `seats_available` and how it was computed but it doesn't seem to do what I'm doing here?
Maybe I'm mistaken let me know, and thanks again for the review!